test/CMakeLists.txt

```diff
@@ -23,8 +23,10 @@ foreach(SRC ${SRCS})
 
   if(${TARGET} MATCHES "^trace_.*")
     target_link_libraries(${TARGET} PRIVATE trace_checker)
+    # ${CMAKE_COMMAND} -E env needed for WILL_FAIL with system level failure
+    # see https://cmake.org/cmake/help/latest/prop_test/WILL_FAIL.html
```

Comment:
@db7 if you would like to not have the `${CMAKE_COMMAND} -E env`, we can change the trace_checker to just exit(1) instead of abort, then it should also be fine